### PR TITLE
feat: Add devise gem

### PR DESCRIPTION
**Pull request by tiendat0139**
## Related Tickets
- [ticket redmine](https://edu-redmine.sun-asterisk.vn/issues/69334)

## WHAT (optional)
- Add devise gem
- Login, logout, remember login use devise
- use devise helper

### CODE REVIEW

**Comment by Huy0110** [comment]
ok

**Comment by tiendat0139** [comment]
ready

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class OrdersController < ApplicationController
-  before_action :logged_in_user, only: %i(index new create)
+  before_action :authenticate_user!
+  before_action :authenticate_user!, only: %i(index new create)
```
Comment (ngocvt-0484) [inline app/controllers/orders_controller.rb:2]: sao lại 2 dòng authenticate_user vậy e

File: db/migrate/20230904224625_create_users.rb [hunk 2]
```diff
@@ -4,13 +4,7 @@ def change
       t.string :first_name
       t.string :last_name
       t.string :email
-      t.string :password_digest
```
Comment (ngocvt-0484) [inline db/migrate/20230904224625_create_users.rb]: tạo file migration mới,. k sửa các file migration cũ e nhé

**Comment by tiendat0139** [comment]
ready

**Pull request merged**